astropy/units/utils.py

```diff
@@ -286,9 +286,11 @@ def resolve_fractions(a, b):
     return a, b
 
 
-def quantity_asanyarray(a, dtype=None):
+def quantity_asanyarray(a, dtype=np.floating):
```

Comment:
This changes the behaviour of this (probably rarely used?) function. As is, any `ndarray` subclasses are passed through, but with your change, they will change their `dtype`. I think the logical thing to do is to keep `None` here, but change any `None` to `np.floating` in the `Quantity` branch. That will keep behaviour exactly identical, which seems the best bet here.

Reply:
✅

Reply:
``quantity_asanyarray`` is basically only used in ``astropy.modeling``. With ``subok=True`` working in ``np.asanyarray``, should this function be deprecated?

Reply:
What's different is that this function recognizes an iterable of `Quantity`. So, I think we better keep it (for now at least).